endpoints/put.go

```diff
@@ -90,7 +90,21 @@ func NewPutHandler(backend backends.Backend, maxNumValues int) func(http.Respons
 			resps.Responses[i].UUID = uuid.NewV4().String()
 			ctx, cancel := context.WithTimeout(context.Background(), 500*time.Millisecond)
 			defer cancel()
-			err = backend.Put(ctx, resps.Responses[i].UUID, toCache, p.TTLSeconds)
+			// Only allow setting a provided key if configured (and ensure a key is provided).
+			if allowKeys && len(p.Key) > 0 {
+				s, err := backend.Get(ctx, p.Key)
```

Comment:
Yes, it is a race condition, but that is a very edge case. To properly handle it, we would need to push the UUID generation code down to each backend and handle it there. That would be a large architecture change just to deal with a very edge case. Or have the backends do an existing key check on every incoming store request, even if was a cache generated UUID (as the backend won't be able to reliably tell who generated the key) which would increase the load on the backend store even further. Not sure this edge case warrants that amount of work.

Reply:
Ultimately that's up to you all.

It _probably_ shouldn't cause problems, since you expect JS clients to be generating UUIDs. On the other hand, this was a thing: https://github.com/prebid/Prebid.js/issues/2822. And race conditions are notoriously hard to debug.

Another issue is that some backends may not even support this as an atomic operation.

@josephveach might have some thoughts here... it's a good excuse to chat about code & project standards.